kernel/src/main/java/org/kframework/kprove/KProve.java

```diff
@@ -55,6 +62,19 @@ public int run() {
 
         Tuple2<Definition, Module> compiled = proofDefinitionBuilder
                 .build(kproveOptions.specFile(files), kproveOptions.defModule, kproveOptions.specModule);
+
+        // Saving compiled definition to disk to be usable by other tools (e.g., kast)
+        CompiledDefinition compiledDefinition = new CompiledDefinition(
+                new KompileOptions(), compiled._1(), compiled._1(), files, kem, null);
+        Path proveKompiledDir = files.resolveTemp("prove-spec-kompiled").toPath();
+        try {
+            Files.createDirectory(proveKompiledDir);
+            loader.saveOrDie(proveKompiledDir.resolve("compiled.bin").toFile(), compiledDefinition);
+        } catch (IOException e) {
+            kem.registerCriticalWarning(
+                    "Could not create prove-spec-kompiled directory. Definition will not be saved.", e);
```

Comment:
It will be useful to log the directory, because I expect there will be multiple dirs there with similar names.

Reply:
if this directory is not writeable it's probably better to raise an error instead of a warning.